### PR TITLE
api: add vim.api.nvim_win_hide

### PR DESCRIPTION
**Pull request by euclidianAce**

### CODE REVIEW

File: src/nvim/ex_docmd.c [hunk 1]
```diff
@@ -6504,6 +6504,23 @@ static void ex_pclose(exarg_T *eap)
   }
 }
 
+/*
+ * Hide window "win"
+ *
+ */
+void
+ex_win_hide(
```
Comment (tjdevries) [inline src/nvim/ex_docmd.c:6512]: Is there anywhere else that should be calling this code?

File: test/functional/api/window_spec.lua [hunk 2]
```diff
@@ -347,4 +347,32 @@ describe('API/win', function()
...
+      local oldwin = meths.get_current_win()
+      local oldbuf = meths.get_current_buf()
+      local buf = meths.create_buf(true, false)
+      local newwin = meths.open_win(buf, true, {
+        relative='win', row=3, col=3, width=12, height=3
+      })
+      meths.win_hide(newwin)
+      eq({oldwin}, meths.list_wins())
+      eq({oldbuf, buf}, meths.list_bufs())
+    end)
```
Comment (tjdevries) [inline test/functional/api/window_spec.lua:376]: can you add a test where you set `bufhidden` for the buffer to be `wipe` and then make sure it wipes it when you hide the window? I think that's the expected behavior.

File: src/nvim/api/window.c [hunk 3]
```diff
@@ -492,6 +492,30 @@ Dictionary nvim_win_get_config(Window window, Error *err)
   return rv;
 }
 
+/// Hides the window (like |:hide| with a |window-ID|).
```
Comment (bfredl) [inline src/nvim/api/window.c:495]: A bit miss-leading. It still closes the window, but the _buffer_ becomes hidden even when 'hidden' not set.

A line or two below quickly summarizing the difference with `nvim_win_close` would be useful.

**Comment by euclidianAce** [comment]
Is there a `make` target to generate the `:help` info? I did `make helptags` but that didn't seem to work. Or is there some CI that takes care of that?

**Comment by tjdevries** [comment]
`./scripts/gen_vimdocs.py` or something like that -- it's not super great atm

**Review by tjdevries — APPROVED** [review]
I think it LGTM -- @bfredl you have any other thoughts?

**Comment by euclidianAce** [comment]
Pinging @bfredl, is there anything else this PR needs?

**Comment by bfredl** [comment]
merged after reverting an incorrect doc update (doxygen heisenbug)

**Pull request merged**